### DISCUSSION TITLE
What is vdplus?

### DISCUSSION THREAD

**Discussion by frosencrantz** [saulpw/visidata#1043] [General]
The 2.5 announcement reports a new private repo **vdplus**.  It is a way to reward Patreon supporters.

Is there any information you can publicly share about this new repo, like what extras it includes.  

Congratulations on the new 2.5 release!  Looks like lot's of great new features!  Thanks for the hard work Saul and Anja.

Also, hopefully  the new repo brings in new sponsors!  The count is up to 35!

**Comment by saulpw** [comment]
Hi @frosencrantz, thanks for asking.  Soon there will be a webpage for it on visidata.org, but it's not quite ready yet.  (But you can take a look at our newly redesigned website already!  It's much easier on the eyes thanks to the hard work of @lkhrs).

These plugins/apis/tools/games are currently available in the vdplus repo:

 ## APIs

- [Google Drive](https://github.com/saulpw/vdplus/tree/develop/api/google): list/open/delete files via [Google Drive API](https://developers.google.com/drive/).
- [Google Sheets](https://github.com/saulpw/vdplus/tree/develop/api/google): load/save new sheet (with `.g` extension) via [Google Sheets API](https://developers.google.com/sheets/).
- [Airtable](https://github.com/saulpw/vdplus/tree/develop/api/airtable): load/save existing table via [Airtable API](https://airtable.com/api).
- [Reddit](https://github.com/saulpw/vdplus/tree/develop/api/reddit): read-only interface to [Reddit API](https://www.reddit.com/dev/api) for subreddits, redditors, submissions, and comments.
- [Zulip](https://github.com/saulpw/vdplus/tree/develop/api/zulip): read-only interface to [Zulip chat API](https://github.com/zulip/zulip).
- [RC Together](https://github.com/saulpw/vdplus/tree/develop/api/rctogether): control an army of bots via [RC Together API](https://docs.rctogether.com/).
- [bit.io](https://github.com/saulpw/vdplus/tree/develop/api/bitio): interface to [bit.io](https://bit.io) API.

### Other tools
- [Web Scraper](https://github.com/saulpw/vdplus/tree/develop/api/scraper): scrape urls from web pages and web pages from urls iteratively and interactively.
- [ping](https://github.com/saulpw/vdplus/tree/develop/api/ping): interactive ping+traceroute
- [top](https://github.com/saulpw/vdplus/tree/develop/api/top): interactive process viewer
- [Galactic Conquest](https://github.com/saulpw/vdplus/tree/develop/api/galcon): a remake of the classic [Galcon](http://www.galcon.com/classic/history.html).

---

In addition, I've also been working on a new library the past few weeks, "dataverter", that can load data in a streaming fashion from many different formats, that allows for a lot of the power of VisiData but with a very simple Python API.  It can autodetect file formats, delimiters, types, as well as filter-on-load (#995).  It's not quite ready yet, but in a short while (I'm on vacation this week) it'll get merged into vdplus, and will be available to any VisiData Plus subscriber.

**Comment by frosencrantz** [comment]
I hope you had a great vacation!

Also, vdplus sounds exciting.  I look forward to the new visidata.org page about this.  It looks like there is one more supporter since I last commented.  

Those descriptions are helpful.  I'll mention I do not have access to any of those links under the vdplus repo.  I think you want some level of openness to help potential supporters understand what they are missing.  Also, is this available to all your Patreon supporters?  It looks like there are multiple levels, but none of the levels mention vdplus, except about an API key.

**Comment by frosencrantz** [comment]
@saulpw has also added the ability to generate SVG graphs via matplotlib.  Mentioned in https://github.com/saulpw/visidata/discussions/1056#discussioncomment-1152256

Saul, it would be great if there was a public single location that listed what is in vdplus.

**Reply by saulpw** [reply]
The vdplus repo is now public, and everything that's in vdplus should be described in the toplevel README: https://github.com/visidata/vdplus

**Reply by frosencrantz** [reply]
That link returns a 404 error.  😞

**Reply by saulpw** [reply]
Oops, I did everything but actually make the repo public :)  It's public now, that link should work.

**Reply by frosencrantz** [reply]
I have access now, even to the source code.  Is that what you were expecting @saulpw ?  This non-sponsoring bloke has access to the vdplus source code.

**Reply by saulpw** [reply]
Yes, I decided to make it available, for everyone's convenience, under a Polyform non-commercial license. This is not an open-source license, but you can use it as an individual or non-profit, even if you are not sponsoring my work. But you would need to sponsor me on Patreon or Github if you want to use it (legally, anyway) in your paid work or at your company.
